### PR TITLE
Performance improvments for daypicker.

### PR DESCRIPTION
**Pull request by Drag13**
Hi guys. Working with Daypicker i have faced some trubles with performance.
scope.rows = this.split(days, 7); caused to re render full list of 42 items. And this cause some troubles with Edge. 
So I want to offer some little fix. It is not great solution, but fixed my problems with performance. 
Before changing whole array of the rows, we are checking is it possible to avoid this.
If first and last day are the same, then we can nly update array insted of changing whole array and re render it.
If you have another solution, this will be great.

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -413,6 +414,28 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
...
+
+      if (!rows || newHash !== daysHash) {
+          daysHash = newHash;
+          return this.split(days, 7);
+      }
+
+      var splitted = this.split(days, 7);
+      for (var i = 0; i < splitted.length; i++) {
+          for (var k = 0; k < splitted[i].length; k++) {
+              angular.merge(rows[i][k], splitted[i][k]);
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: This is a potentially very expensive operation

File: src/datepicker/datepicker.js [hunk 2]
```diff
@@ -413,6 +414,28 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
...
+          }
+      }
+
+      return rows;
+  }
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Need semi-colon at the end of this line

**Comment by Drag13** [comment]
Yeh, merge  operation is really expensive. I thought about this side of the fix also. But re-rendering 42 elements with creating new scopes also is not chip. Ill add semicolumn and think. Possibly I can use merge that already exists upper in the code.

**Comment by Drag13** [comment]
12 tests faild. Not great.

**Comment by icfantv** [comment]
@Drag13, do you have an update on this?  If not, I'm going to close it.  Thanks.

**Comment by Drag13** [comment]
I tried some ideas but nothing good enough for me and test. So close. If any Ill ping. 
Have a nice day/

**Comment by icfantv** [comment]
Closing per comment.

**Pull request closed without merging**